src/main/java/org/wildfly/security/password/impl/OneTimePasswordImpl.java

```diff
@@ -59,6 +70,17 @@
         this(algorithm, spec.getHash().clone(), spec.getSeed().clone(), spec.getSequenceNumber());
     }
 
+    OneTimePasswordImpl(final String algorithm, final char[] password, final OneTimePasswordAlgorithmSpec spec) throws SaslException {
+        this(algorithm,
+                generateOTP(algorithm,
+                        getNormalizedPasswordBytes(password),
+                        new String(spec.getSeed()).toLowerCase(Locale.ENGLISH).getBytes(StandardCharsets.US_ASCII),
```

Comment:
This seems wrong.  I think we should consider making the seed be a String instead of a `byte[]`.  According to RFC 2289:

>    The seed MUST consist of purely alphanumeric characters and MUST be
   of one to 16 characters in length. The seed is a string of characters
   that MUST not contain any blanks and SHOULD consist of strictly
   alphanumeric characters from the ISO-646 Invariant Code Set.  The
   seed MUST be case insensitive and MUST be internally converted to
   lower case before it is processed.

Since this value must always be textual in nature, it seems wrong to represent it as a `byte[]`.  @fjuma is there a justification for this or can we make this field be a String?

A String has a few advantages like immutability (one less array to copy) and ease of manipulation.